auth/credentials/impersonate/idtoken.go

```diff
@@ -108,48 +108,39 @@ func NewIDTokenCredentials(opts *IDTokenOptions) (*auth.Credentials, error) {
 			}
 		}
 		client, err = httptransport.NewClient(&httptransport.Options{
-			Credentials: creds,
-			Logger:      logger,
+			Credentials:    creds,
+			UniverseDomain: opts.UniverseDomain,
+			Logger:         logger,
 		})
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	universeDomainProvider := resolveUniverseDomainProvider(creds)
 	itp := impersonatedIDTokenProvider{
-		client:          client,
-		targetPrincipal: opts.TargetPrincipal,
-		audience:        opts.Audience,
-		includeEmail:    opts.IncludeEmail,
-		logger:          logger,
+		client: client,
+		// Pass the credentials universe domain provider to configure the endpoint.
+		universeDomainProvider: universeDomainProvider,
+		targetPrincipal:        opts.TargetPrincipal,
+		audience:               opts.Audience,
+		includeEmail:           opts.IncludeEmail,
+		logger:                 logger,
 	}
 	for _, v := range opts.Delegates {
-		itp.delegates = append(itp.delegates, formatIAMServiceAccountName(v))
+		itp.delegates = append(itp.delegates, internal.FormatIAMServiceAccountResource(v))
 	}
 
-	var udp auth.CredentialsPropertyProvider
-	if creds != nil {
-		udp = auth.CredentialsPropertyFunc(creds.UniverseDomain)
-	}
 	return auth.NewCredentials(&auth.CredentialsOptions{
 		TokenProvider:          auth.NewCachedTokenProvider(itp, nil),
-		UniverseDomainProvider: udp,
+		UniverseDomainProvider: universeDomainProvider,
 	}), nil
 }
 
-type generateIDTokenRequest struct {
-	Audience     string   `json:"audience"`
-	IncludeEmail bool     `json:"includeEmail"`
-	Delegates    []string `json:"delegates,omitempty"`
-}
-
-type generateIDTokenResponse struct {
-	Token string `json:"token"`
-}
-
 type impersonatedIDTokenProvider struct {
-	client *http.Client
-	logger *slog.Logger
+	client                 *http.Client
```

Comment:
Same with my other comment, this structure should go away in favor of the common code